src/Features/Core/AddMissingReference/AbstractAddMissingReferenceCodeFixProvider.cs

```diff
@@ -95,18 +96,18 @@ private static IEnumerable<ITypeSymbol> GetTypesFromSymbol(ISymbol symbol)
         /// so it is safe to assume if this case exists for one of the symbols given 
         /// it is the assembly we want to add.
         /// </summary>
-        private static AssemblyIdentity GetAssemblyIdentity(IEnumerable<ITypeSymbol> types)
+        private static AssemblyIdentity GetAssemblyIdentity(IEnumerable<ITypeSymbol> types, string message)
         {
             foreach (var type in types)
             {
                 var identity = type?.GetBaseTypesAndThis().OfType<IErrorTypeSymbol>().FirstOrDefault()?.ContainingAssembly?.Identity;
-                if (identity != null)
+                if (identity != null && message.Contains(identity.ToString()))
```

Comment:
Wasn't message parsing the entire thing we were trying to get out of in the first place?